From fc85dc819da6dcc83bc6da0bb01e725256d2977e Mon Sep 17 00:00:00 2001
From: "Roberto E. Vargas Caballero" <k0ga@shike2.com>
Date: Fri, 22 Sep 2023 14:18:14 +0200
Subject: [PATCH 6/7] xargs: Free argument before substitution

strnsubst() allocates a new array always, indepently of
the number of occurences of the replace string.
---
 xargs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xargs.c b/xargs.c
index 30b09ab..0d37ec8 100644
--- a/xargs.c
+++ b/xargs.c
@@ -266,9 +266,11 @@ main(int argc, char *argv[])
 				argsz += arglen + 1;
 			} else {
 				for (j = 1; j < i; j++) {
+					char *p = cmd[j];
 					argsz -= strlen(cmd[j]);
 					strnsubst(&cmd[j], replstr, arg, 255);
 					argsz += strlen(cmd[j]);
+					free(p);
 				}
 			}
 
-- 
2.43.0

